pkgs/development/tools/hover/default.nix

```diff
@@ -0,0 +1,72 @@
+{ buildGoPackage, fetchFromGitHub, fetchurl, glfw, libGL, libX11, libXcursor, libXi,
+  libXinerama, libXrandr, libXxf86vm, makeWrapper, pkgconfig, stdenv, unzip}:
+
+buildGoPackage rec {
```

Comment:
please use buildGoModule, upstream has a go.mod

https://nixos.org/nixpkgs/manual/#ssec-go-modules

Reply:
I tried, but it looks like it can't build some submodule :disappointed: 

> these derivations will be built:                                                                                                                                                                           
  /nix/store/xadpp25aznchvd8y2316fc9ig9bgx6s5-hover-unstable-2020-01-23.drv                                                                                                                                
building '/nix/store/xadpp25aznchvd8y2316fc9ig9bgx6s5-hover-unstable-2020-01-23.drv'...
unpacking sources                                                                                                                                                                                         
unpacking source archive /nix/store/sg2545v1p8vfdaf8m2bivx2h2cqis0i4-source                                                                                                                             
source root is source                                                                                                                                                                                    
patching sources                                                                                                                                                                                           
applying patch /nix/store/9zdn36yhwv1lkr97hb0dl0n8lp1vwlfn-fix-assets-path.patch                                                                                                                           
patching file internal/fileutils/assets.go                                                                                                                                                                 
patching file internal/fileutils/file.go                                                                                                                                                                   
applying patch /nix/store/8cgz7xa735a85ywcqlmqx4kd4dcffnck-fix-build-docker.patch                                                                                                                          
patching file cmd/build.go                                                                                                                                                                                 
configuring                                                                                                                                                                                                
preBuildPhase                                                                                                                                                                                             
Archive:  /nix/store/5hdd7rwy9gaxgq163xx47rf30d0a1aw9-libflutter_engine.zip                                                                                                                               
  inflating: flutter_embedder.h                                                                                                                                                                           
  inflating: libflutter_engine.so                                                                                                                                                                         
building                                                                                                                                                                                                  
Building subPackage .                                                                                                                                                                                     
go: downloading github.com/pkg/errors v0.9.1                                                                                                                                                                
go: downloading github.com/otiai10/copy v1.0.2                                                                                                                                                             
go: downloading gopkg.in/yaml.v2 v2.2.8                                                                                                                                                                    
go: downloading github.com/tcnksm/go-latest v0.0.0-20170313132115-e3007ae9052e                                                                                                                             
go: extracting github.com/otiai10/copy v1.0.2                                                                                                                                                              
go: extracting github.com/tcnksm/go-latest v0.0.0-20170313132115-e3007ae9052e                                                                                                                              
go: downloading github.com/logrusorgru/aurora v0.0.0-20200102142835-e9ef32dff381                                                                                                                           
go: downloading github.com/spf13/cobra v0.0.5                                                                                                                                                              
go: downloading github.com/hashicorp/go-version v1.2.0                                                                                                                                                     
go: extracting github.com/hashicorp/go-version v1.2.0
go: downloading golang.org/x/net v0.0.0-20191101175033-0deb6923b6d9
go: downloading github.com/google/go-github v17.0.0+incompatible
go: extracting github.com/pkg/errors v0.9.1
go: extracting gopkg.in/yaml.v2 v2.2.8
go: extracting github.com/logrusorgru/aurora v0.0.0-20200102142835-e9ef32dff381
go: extracting github.com/spf13/cobra v0.0.5
go: downloading github.com/spf13/pflag v1.0.5
go: extracting github.com/spf13/pflag v1.0.5
go: extracting github.com/google/go-github v17.0.0+incompatible
go: downloading github.com/google/go-querystring v1.0.0
go: extracting github.com/google/go-querystring v1.0.0
go: extracting golang.org/x/net v0.0.0-20191101175033-0deb6923b6d9
go: finding github.com/logrusorgru/aurora v0.0.0-20200102142835-e9ef32dff381
go: finding github.com/pkg/errors v0.9.1
go: finding gopkg.in/yaml.v2 v2.2.8
go: finding github.com/otiai10/copy v1.0.2
go: finding github.com/tcnksm/go-latest v0.0.0-20170313132115-e3007ae9052e
go: finding github.com/google/go-github v17.0.0+incompatible
go: finding github.com/google/go-querystring v1.0.0
go: finding github.com/hashicorp/go-version v1.2.0
go: finding golang.org/x/net v0.0.0-20191101175033-0deb6923b6d9
go: finding github.com/spf13/cobra v0.0.5
go: finding github.com/spf13/pflag v1.0.5
github.com/logrusorgru/aurora
github.com/go-flutter-desktop/hover/internal/log
github.com/go-flutter-desktop/hover/internal/androidmanifest
github.com/go-flutter-desktop/hover/internal/build
runtime/cgo
net
github.com/go-flutter-desktop/hover/internal/fileutils
github.com/pkg/errors
gopkg.in/yaml.v2
github.com/go-flutter-desktop/hover/internal/pubspec
github.com/otiai10/copy
os/user
github.com/go-flutter-desktop/hover/cmd/packaging
github.com/go-flutter-desktop/hover/internal/config
github.com/go-flutter-desktop/hover/internal/enginecache
github.com/google/go-querystring/query
github.com/google/go-github/github
github.com/hashicorp/go-version
golang.org/x/net/html/atom
golang.org/x/net/html
github.com/tcnksm/go-latest
github.com/go-flutter-desktop/hover/internal/versioncheck
github.com/spf13/pflag
github.com/spf13/cobra
github.com/go-flutter-desktop/hover/cmd
github.com/go-flutter-desktop/hover
Building subPackage ./assets/app
go: directory assets/app is outside main module
builder for '/nix/store/xadpp25aznchvd8y2316fc9ig9bgx6s5-hover-unstable-2020-01-23.drv' failed with exit code 1
error: build of '/nix/store/xadpp25aznchvd8y2316fc9ig9bgx6s5-hover-unstable-2020-01-23.drv' failed

Reply:
interesting, I'm not familiar enough with go to help :(

Reply:
@zimbatm @kalbasit

Reply:
me too :cry: 
I'm completely open for suggestions

Reply:
It seems to have a sub-module under https://github.com/go-flutter-desktop/hover/tree/master/assets/app and `buildGoModule` cannot handle sub-modules yet. The workaround here is to build only the root of the project using `subPackages = [ "." ];`.

Reply:
Done, thanks @jonringer and @kalbasit !